command/volume_create_host.go

```diff
@@ -19,13 +19,34 @@ import (
 )
 
 func (c *VolumeCreateCommand) hostVolumeCreate(
-	client *api.Client, ast *ast.File, detach, verbose, override bool) int {
+	client *api.Client, ast *ast.File, detach, verbose, override bool, volID string) int {
 
 	vol, err := decodeHostVolume(ast)
 	if err != nil {
 		c.Ui.Error(fmt.Sprintf("Error decoding the volume definition: %s", err))
 		return 1
 	}
+	if volID != "" {
+		ns := c.namespace
+		if vol.Namespace != "" {
+			ns = vol.Namespace
+		}
```

Comment:
are these things simultaneously true?

* namespace in the volume spec will override namespace in the CLI
* volume ID in the CLI will override volume ID in the volume spec

if so, I get some cognitive dissonance. though I suppose it has to be this way, because there's a default namespace in the CLI (flag or env), but of course no such default for volume IDs.

Reply:
Yes! But yeah, that's just the way is has to be because there's such a thing as a default namespace, whereas the "default ID" is the server-generated unique ID.